File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        int M = scanner.nextInt();
        
        int[] basket = new int[N];
        for(int i = 0; i < N; i++)
        {
            basket[i] = i + 1;
        }
        
        for(int t = 0; t < M; t++)
        {
            int i = scanner.nextInt();
            int j = scanner.nextInt();
            int left = i - 1;
            int right = j - 1;
            while(left < right)
            {
                int temp = basket[left];
                basket[left] = basket[right];
                basket[right] = temp;
                left++;
                right--;
            }
        }
       
        for (int num : basket) {
            System.out.print(num + " ");
        }
        scanner.close();
    }
}

File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.cc
#include <stdio.h>

int main()
{
    int N, M, I, J;
    scanf("%d %d", &N, &M);
    int array[101];

    for(int a = 0; a < N; a++)
    {
        array[a] = a + 1;
    }
    
  
    for(int b = 0; b < M; b++)
    {
        scanf("%d %d", &I, &J);
        int left = I - 1;
        int right = J - 1;
        while(left < right)
        {
            int temp = array[left];
            array[left] = array[right];
            array[right] = temp;
            left++;
            right--;
        }
    }

    for(int a = 0; a < N; a++)
    {
        printf("%d ", array[a]);
    }

    return 0;
}